project-koku/masu
# OCP local ingesting and processing stubs

**Overview**
A local ingestion path for OCP is needed to enable development work without the dependency of the application platform.  This is valuable because those bits of how the platform will provide usage data are still under development and also we do not need to have the platform in place for report processing and orchestration work.

Much like that for AWS-local, OCP-local provides a way to specify a local folder to stand in for the application platform's upload service.  The directory that masu is relying on for this is `/var/tmp/insights_local`.  A .csv file is placed in a subdirectory for the format `<cluster-id>/<month date range>/report.csv

Here's an example:
```
$ ls /var/tmp/insights_local/my-ocp-cluster-1/20180901-20181001/
pod-cpu-usage-aws.csv
```

Downloaded files will reside in the masu temporary file directory.  Example:
```
$ ls /var/tmp/masu/TestCustomer/ocp-local/my-ocp-cluster-1/
20180901-20181001_pod-cpu-usage-aws.csv
```
Just like that for AWS, an OCP-specific report processor will be implemented and will work with files in this temporary directory.  This change introduces a stub for that processor.

**Future Work**
* Implement the `OCPReportProcessor` methods for processing, removing temporary processed files, and removing expired report data.

**Testing**
1. Setup an OCP-local provider
2.  Add a .csv file into the local storage area
3. Download/process with masu download endpoint
4. Verify temporary storage directory 

Testing Results:

[masu_ocp_local.txt](https://github.com/project-koku/masu/files/2413235/masu_ocp_local.txt)

Closes #197
